extensions/amp-gfycat/0.1/amp-gfycat.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+import {user} from '../../../src/log';
+
+class AmpGfycat extends AMP.BaseElement {
+
+  /** @override */
+  preconnectCallback(onLayout) {
+    // Gfycat iframe
+    this.preconnect.url('https://gfycat.com', onLayout);
+
+    // Iframe video and poster urls
+    this.preconnect.url('https://giant.gfycat.com', onLayout);
+    this.preconnect.url('https://thumbs.gfycat.com', onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const gfyid = user().assert(
+      (this.element.getAttribute('data-gfyid') ||
+      this.element.getAttribute('gfyid')),
+      'The data-gfyid attribute is required for <amp-gfycat> %s',
+      this.element);
+    const width = this.element.getAttribute('width');
+    const height = this.element.getAttribute('height');
+    const layoutType = this.element.getAttribute('layout');
+    const autoplay = parseInt(this.element.getAttribute('data-autoplay') ||
```

Comment:
we have it like this because video is autoplaying by default, so we needed `autoplay=0` to turn it off
if we just use `autoplay`, video will be paused by default?